packages/surveycto/src/Adaptor.js

```diff
@@ -24,92 +21,133 @@ export function execute(...operations) {
   };
 
   return state => {
-    state.configuration.baseUrl = 'https://'.concat(
-      state.configuration.instanceName,
-      '.surveycto.com/api/v1'
-    );
     return commonExecute(...operations)({ ...initialState, ...state });
   };
 }
 
 /**
- * Make a GET request and POST it somewhere else
- * @example
- * execute(
- *   fetch(params)
- * )(state)
+ * Options provided to the HTTP request
+ * @typedef {Object} FormSubmissionOptions
+ * @property {string} [date=0] - Form completion or submission date. Default to `0` which will return all submission data
+ * @property {string} [format='json'] - Form response type, It can be in `csv` or `json`. Default to `json`
+ * @property {string} status - (Opt)Review status. Can be either, `approved`, `rejected`, `pending` or combine eg `approved|rejected`.
+ */
+
+/**
+ * Fetch form submissions
+ * @example <caption>Using state lazy load</caption>
+ * fetchSubmissions($.formId || 'test', { date: '2024-04-18' });
+ * @example <caption> With huma readable date</caption>
+ * fetchSubmissions('test', { date: 'Apr 18, 2024 6:26:21 AM' });
+ * @example <caption> Formatting the results to CSV String</caption>
+ * fetchSubmissions('test', { date: '2024-04-20', format: 'csv' });
+ * @example <caption> With reviewStatus filter</caption>
+ * fetchSubmissions('test', {
+ *   date: '2024-04-18',
+ *   status: 'approved|rejected',
+ * });
+ * @example <caption> With access to the callback</caption>
+ * fetchSubmissions(
+ *   'test',
+ *   {
+ *     date: 'Apr 18, 2024 6:26:21 AM',
+ *     status: 'approved|rejected',
+ *   },
+ *   state => {
+ *     console.log('Hello from the callback!');
+ *     return state;
+ *   }
+ * );
  * @function
- * @param {object} params - data to make the fetch
+ * @param {string} formId - Form id
+ * @param {FormSubmissionOptions} options - Form submission date, timestamp, format, status parameters
```

Comment:
Date needs better documentation than "form submission date". For one thing, the key is `options.date`, which is only referenced in examples.

I think it's really important to make it clear that you can pass any anything parseable by ` new Date(dateString)` and it will be converted into second resolution for the surveyCTO surver. Because that's the contract we agreed early and that's what the internal API expects.

I guess it's an implementation detail, but anyone who expects to pass a surveyCTO date in seconds is going to get a nasty shock.

Btw can you pass an actual date object? Should we support that?

Reply:
@mtuchi just link to the REST API docs (link below) and say the following: 
```
Date (request) parameters should be specified in [UTC time](https://en.wikipedia.org/wiki/Coordinated_Universal_Time). Bypass and request all data by specifying `date=0`.
See API docs: https://support.surveycto.com/hc/en-us/articles/360033156894-REST-API-documentation
```

And you can add the below from their docs if you want more examples: 
```
For example, in order to fetch data after midnight Oct. 12 2015, you can use a URL like one of the following:

https://servername.surveycto.com/api/v2/forms/data/wide/json/formid?date=1444694400

https://servername.surveycto.com/api/v2/forms/data/wide/json/formid?date=1444694400000
```

Reply:
I have removed the date conversion, Now you can pass a either `unix timestamp` in seconds/milliseconds or a date time with this format `MMM dd, yyyy h:mm:ss a`
cc @aleksa-krolls @josephjclark